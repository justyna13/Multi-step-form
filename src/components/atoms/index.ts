export { FormStep } from "@/components/atoms/FormStep/FormStep.tsx";
export { Button } from "@/components/atoms/Button/Button.tsx";
export { Input } from "@/components/atoms/Input/Input.tsx";
export { Header } from "@/components/atoms/Header/Header.tsx";
