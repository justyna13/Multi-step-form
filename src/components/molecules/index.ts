export { FormNavigation } from "@/components/molecules/FormNavigation/FormNavigation.tsx";
export { Footer } from "@/components/molecules/Footer/Footer.tsx";
export { FormInput } from "@/components/molecules/FormInput/FormInput.tsx";
export { FormSummary } from "@/components/molecules/FormSummary/FormSummary.tsx";
