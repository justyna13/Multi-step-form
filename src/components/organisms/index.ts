export { FormStepFirst } from '@/components/organisms/FormStepFirst/FormStepFirst.tsx';
export { FormStepSecond } from '@/components/organisms/FormStepSecond/FormStepSecond.tsx';
export { FormStepThird } from '@/components/organisms/FormStepThird/FormStepThird.tsx';
export { FormStepFourth } from '@/components/organisms/FormStepFourth/FormStepFourth.tsx';
export { FormConfirmation } from '@/components/organisms/FormConfirmation/FormConfirmation.tsx';
