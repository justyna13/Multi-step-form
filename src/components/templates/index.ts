export { HomePageTemplate } from '@/components/templates/HomePageTemplate/HomePageTemplate.tsx';
