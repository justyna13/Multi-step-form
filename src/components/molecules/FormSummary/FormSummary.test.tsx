import { render } from '@testing-library/react';
import { describe, expect } from 'vitest';

import { FormSummary } from '@/components/molecules';
import {
  IFormProvider,
  MultiStepFormProvider
} from '@/store/multiStepForm/MultiStepFormProvider.tsx';

const formWrapper = ({ children }: IFormProvider) => (
  <MultiStepFormProvider>{children}</MultiStepFormProvider>
);

describe('FormSummary', () => {
  test('should match snapshot', () => {
    const view = render(<FormSummary />, { wrapper: formWrapper });

    expect(view).toMatchSnapshot();
  });
});
