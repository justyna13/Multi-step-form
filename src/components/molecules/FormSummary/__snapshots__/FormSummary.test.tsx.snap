// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormSummary > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-summary"
        data-testid="form-summary"
      >
        <div
          class="form-summary__inner"
        >
          <div
            class="form-summary__plan"
          >
            <div>
              <p>
                 (Monthly)
              </p>
              <span
                class="link"
              >
                Change
              </span>
            </div>
            <span>
              +undefined/mo
            </span>
          </div>
        </div>
        <div
          class="form-summary__total"
        >
          <p>
            Total (per month/year)
          </p>
          <span>
            +$0/mo
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-summary"
      data-testid="form-summary"
    >
      <div
        class="form-summary__inner"
      >
        <div
          class="form-summary__plan"
        >
          <div>
            <p>
               (Monthly)
            </p>
            <span
              class="link"
            >
              Change
            </span>
          </div>
          <span>
            +undefined/mo
          </span>
        </div>
      </div>
      <div
        class="form-summary__total"
      >
        <p>
          Total (per month/year)
        </p>
        <span>
          +$0/mo
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
